src/content/en/fundamentals/native-hardware/capturing-images/index.md

```diff
@@ -2,210 +2,343 @@ project_path: /web/_project.yaml
+the `drop` event;
 
-* Attach it directly to a `<canvas>` element so that you can manipulate it
-* Download it to the user's device
-* Upload it to a server by attaching to an `XMLHttpRequest` 
+The `dragover` event handler let's us signal to the user what will happen when they drop the file by
+using [the dropEffect property](https://developer.mozilla.org/en-US/docs/Web/API/DataTransfer/dropEffect).
 
-Whilst using the input element method of getting access to images is 
-ubiquitous, it is the least appealing option because it is not integrated 
-directly into the web page, and on desktop can't access the user's webcam.
+Drag and drop has been around for a long time and is well supported by the major browsers.
+
+### Paste from clipboard
+The final way to get an existing image file is from the clipboard. The code for this is very simple,
+but the user experience is a little harder to get right.
+
+    <textarea id="target">Paste an image here</textarea>
+    <script>
+      const target = document.getElementById('target');
+
+      target.addEventListener('paste', (e) => {
+        e.preventDefault();
+        doSomethingWithFiles(e.clipboardData.files);
+      });
+    </script>
+
+(`e.clipboardData.files` is yet another `FileList` object.)
+
+The tricky part with the clipboard API is that, for full cross-browser support, the target element
+needs to be both selectable and editable. Both `<textarea>` and `<input type="text">` fit the bill
+here, as do elements with the `contenteditable` attribute, but these are also obviously designed for
+editing text.
+
+It can be difficult to make this work smoothly if you don't want to user to be able to input
+text. Tricks like having a hidden input that gets selected when you click on some other element may
+make maintaining accessibility harder.
+
+### Handling a FileList object
+Since most of the above methods produce a `FileList`, we should talk a little about what that is.
+
+A `FileList` is similar to an `Array`. It has numeric keys and a `length` property, but it isn't
+*actually* an array. There are no array methods, like `forEach` or `pop`, and it isn't iterable.
+Of course, you can get a real Array by using `Array.from(fileList)`.
+
+The entries of the `FileList` are `File` objects. These are exactly the same as `Blob` objects
+except that they have additional `name` and `lastModified` read-only properties.
+
+    <img id="output">
+    <script>
+      const output = document.getElementById('output');
+
+      function doSomethingWithFiles(fileList) {
+        let file = null;
+
+        for (let i = 0; i < fileList.length; i++) {
+          if (fileList[i].type.match(/^image\//)) {
+            file = fileList[i];
+            break;
+          }
+        }
+
+        if (file !== null) {
+          output.src = URL.createObjectURL(file);
+        }
+      }
+    </script>
+
+This example finds the first file that has an image MIME type, but you could also handle multiple
```

Comment:
'but you could also'  Should this say 'but it could also'. If you're not going to show that, could you give a clue as to how it's done? Is it as simple as changing 

```
file = fileList[i];
break;
```

to 

```
imagesFiles.push(fileList[i]);
```